### PR TITLE
[serve] Improve access log messages

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

Improves the Serve access log to include HTTP status information on the replica and better resemble standard log formats like `uvicorn`.

I'm updated the log format to:

- Include the HTTP method, route and status code for HTTP requests like: "GET / 200".
- Use "CALL", the method name, and "OK"/"ERROR" for `DeploymentHandle` calls like: "CALL method_name OK".
- Stop logging the `route` information on every message and isolate it to the access log message instead. This adds clutter and it can be tied to the access log message using the request ID.
- Stop logging the filename and line number in every log message. This has been on my TODO list for awhile after internal discussion.

### Examples

Calling a handle method:
```
INFO 2024-11-20 13:51:44,803 default_D lswbv98w d0cadb74-9fd8-4a68-9e9a-c9ed20b091f4 -- CALL method OK 1.3ms
```

Calling a basic HTTP route:
```
INFO 2024-11-20 13:53:00,197 default_A zosusx8c 26891433-a91a-4d27-b543-0936feb5f5c1 -- GET / 200 4.1ms
```

Calling a wildcard HTTP route (and non-200 status code):
```
INFO 2024-11-20 13:53:58,102 default_A iovmsu5e 85000f14-8e31-42a5-a1f4-0fa2d39c549a -- GET /{wildcard} 422 3.7ms
```

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/_private/replica.py [hunk 1]
```diff
@@ -430,14 +453,16 @@ def _wrap_user_method_call(
 
         logger.info(
             access_log_msg(
-                method=request_metadata.call_method,
-                status=status_str,
+                method=http_method or "CALL",
```
Comment (GeneDer) [inline python/ray/serve/_private/replica.py:456]: Non-blocker nit: Can probably refactor a constant for the default method 🙃

**Review by GeneDer — APPROVED** [review]
Looks great!

File: python/ray/serve/_private/replica.py [hunk 2]
```diff
@@ -508,19 +559,24 @@ async def handle_request_with_rejection(
...
+            request_metadata, request_args
+        ) as status_code_callback:
+            yield pickle.dumps(
+                ReplicaQueueLengthInfo(
+                    accepted=True,
+                    # NOTE(edoakes): `_wrap_user_method_call` will increment the number
+                    # of ongoing requests to include this one, so re-fetch the value.
+                    num_ongoing_requests=self.get_num_ongoing_requests(),
+                )
             )
```
Comment (zcin) [inline python/ray/serve/_private/replica.py:572]:
```suggestion
            yield ReplicaQueueLengthInfo(
                accepted=True,
                # NOTE(edoakes): `_wrap_user_method_call` will increment the number
                # of ongoing requests to include this one, so re-fetch the value.
                num_ongoing_requests=self.get_num_ongoing_requests(),
            )
```
I think with the new interface this is pickled in `ReplicaActor`

**Review by zcin — APPROVED** [review]
overall lgtm!

File: python/ray/serve/_private/replica.py (hunk 2 above)
Comment (edoakes) [reply python/ray/serve/_private/replica.py:572]: oh good call!

**Pull request merged**